### PR TITLE
DRILL-7119: Compute range predicate selectivity using histograms.

### PR DESCRIPTION
**Pull request by amansinha100**

### CODE REVIEW

**Comment by amansinha100** [comment]
@gparai could you pls review ?

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/NumericEquiDepthHistogram.java [hunk 1]
```diff
@@ -69,27 +79,177 @@ public void setNumRowsPerBucket(long numRows) {
...
+
+            // if value is less than or equal to the first bucket's start point then all rows qualify
+            int result = value.compareTo(buckets[first]);
+            if (result <= 0) {
+              return LARGE_SELECTIVITY;
+            }
+            // if value is greater than the end point of the last bucket, then none of the rows qualify
+            result = value.compareTo(buckets[last]);
+            if (result > 0) {
+              return SMALL_SELECTIVITY;
```
Comment (gparai) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/NumericEquiDepthHistogram.java]: For posterity, please add an explanation for SMALL_SELECTIVITY > 0 and the meaning of buckets[n], if not added elsewhere.

**Comment by gparai** [comment]
@amansinha100 except the minor comment the code looks good. I do not see a testcase illustrating the use of histograms.

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/NumericEquiDepthHistogram.java (hunk 1 above)
Comment (amansinha100) [reply exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/NumericEquiDepthHistogram.java]: Done.  I added a comment when we declare the class variable.

**Comment by amansinha100** [comment]
@gparai I added a unit test for histogram usage and addressed your comment.  Could you pls take another look ?

**Pull request merged**